tests/light/functional_tests/source_drivers/network_source/proxyprotocol/test_pp_network.py

```diff
@@ -0,0 +1,117 @@
+            flags="no-parse",
+        )
+
+    if (template):
+        file_destination = config.create_file_destination(file_name=output_file, template=template)
+    else:
+        file_destination = config.create_file_destination(file_name=output_file)
+    config.create_logpath(statements=[network_source, file_destination])
+
+    syslog_ng.start(config)
+    if password is not None:
+        syslog_ng_ctl.credentials_add(credential=server_key_path, secret=password)
+
+    loggen_input_file_path = Path("loggen_input_{}.txt".format(get_unique_id()))
+    loggen_input_file = File(loggen_input_file_path)
+    loggen_input_file.write_content_and_close(input_messages)
+    loggen.start(
+        network_source.options["ip"], network_source.options["port"], number=number_of_messages,
+        inet=use_inet,
+        use_ssl=use_ssl,
+        read_file=str(loggen_input_file_path),
+        dont_parse=True,
+        proxied=1,
+        proxied_tls_passthrough=use_passthrough,
+        proxy_src_ip="1.1.1.1", proxy_dst_ip="2.2.2.2", proxy_src_port="3333", proxy_dst_port="4444",
+    )
+    # seems proxy header is counted in get_sent_message_count(), so we need '-1'
+    wait_until_true(lambda: loggen.get_sent_message_count() == number_of_messages - 1)
+
+    assert file_destination.read_log() == expected_messages
+
+
+def test_pp_network_tcp(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters):
+    TEMPLATE = r'"${PROXIED_SRCIP} ${PROXIED_DSTIP} ${PROXIED_SRCPORT} ${PROXIED_DSTPORT} ${PROXIED_IP_VERSION} ${MESSAGE}\n"'
+    INPUT_MESSAGES = "message 0\n"
+    EXPECTED_MESSAGES = "1.1.1.1 2.2.2.2 3333 4444 4 message 0\n"
+    NUMBER_OF_MESSAGES = 2
+    _test_pp(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters, '"proxied-tcp"', INPUT_MESSAGES, NUMBER_OF_MESSAGES, EXPECTED_MESSAGES, TEMPLATE)
+
+
+def test_pp_network_tls(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters):
+    TEMPLATE = r'"${PROXIED_SRCIP} ${PROXIED_DSTIP} ${PROXIED_SRCPORT} ${PROXIED_DSTPORT} ${PROXIED_IP_VERSION} ${MESSAGE}\n"'
+    INPUT_MESSAGES = "message 0\n"
+    EXPECTED_MESSAGES = "1.1.1.1 2.2.2.2 3333 4444 4 message 0\n"
+    NUMBER_OF_MESSAGES = 2
+    _test_pp(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters, '"proxied-tls"', INPUT_MESSAGES, NUMBER_OF_MESSAGES, EXPECTED_MESSAGES, TEMPLATE)
+
+
+def test_pp_network_tls_with_passphrase(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters):
+    TEMPLATE = r'"${PROXIED_SRCIP} ${PROXIED_DSTIP} ${PROXIED_SRCPORT} ${PROXIED_DSTPORT} ${PROXIED_IP_VERSION} ${MESSAGE}\n"'
+    INPUT_MESSAGES = "message 0\n"
+    EXPECTED_MESSAGES = "1.1.1.1 2.2.2.2 3333 4444 4 message 0\n"
+    NUMBER_OF_MESSAGES = 2
+    _test_pp(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters, '"proxied-tls"', INPUT_MESSAGES, NUMBER_OF_MESSAGES, EXPECTED_MESSAGES, TEMPLATE, password="asdfg")
```

Comment:
We should use a different set of certs in this case:
```py
    server_key_path = copy_shared_file(testcase_parameters, "server-protected-asdfg.key")
    server_cert_path = copy_shared_file(testcase_parameters, "server-protected-asdfg.crt")
```

Same for the syslog tests.